Challenge2/app.js

```diff
@@ -0,0 +1,128 @@
+} else {
+    ready()
+}
+
+function ready() {
+    var quantityEL = document.getElementsByClassName('quantity');
+    var decreaseCartItemButtons = document.getElementsByClassName("decrease");
+    var increaseCartItemButtons = document.getElementsByClassName("increase");
+    for (var i = 0; i < quantityEL.length; i++) {
+        var input = quantityEL[i]
+        increaseCartItemButtons[i].addEventListener('click',increaseCartItem)
+        decreaseCartItemButtons[i].addEventListener('click',decreaseCartItem)
+    }
+
+    var addToCartButtons = document.getElementsByClassName('add')
+    for (var i = 0; i < addToCartButtons.length; i++) {
+        var button = addToCartButtons[i]
+        button.addEventListener('click', addToCartClicked)
+    }
+}
+
+function increaseCartItem(event){
+    let buttonClicked  = event.target;
+    let buttonEL = buttonClicked.parentElement.parentElement.getElementsByClassName("quantity");
+    buttonEL[0].innerText = parseInt(buttonEL[0].innerText) + 1;
+    updateCartTotal()
+}
+function decreaseCartItem(event){
+    let buttonClicked  = event.target;
+    let buttonEL = buttonClicked.parentElement.parentElement.getElementsByClassName("quantity");
+    buttonEL[0].innerText = parseInt(buttonEL[0].innerText) - 1;
+    if(buttonEL[0].innerText == 0) {
+        var id = event.path[1].name;
+        console.log(event)
+        buttonClicked.parentElement.parentElement.parentElement.remove()
+
+        let buttonToBeReplaced = document.getElementById(id).parentElement;
+        // buttonToBeReplaced.remove();
+        let addToCartButtonContents = `<button id="${id}" class="add">Add to Cart</button>`;
+        let addToCartButton = document.createElement("add");
+        addToCartButton.innerHTML = addToCartButtonContents;
+        buttonToBeReplaced.parentNode.replaceChild(addToCartButton, buttonToBeReplaced);
+        console.log(buttonToBeReplaced)
+        // buttonToBeReplaced.addEventListener('click', addToCartClicked)
+    }
+    updateCartTotal()
+}
+
+
+function addToCartClicked(event) {
+    let button = event.target;
+    let menuItem = button.parentElement.parentElement;
+    let title = menuItem.getElementsByClassName("menu-item")[0].innerHTML;
+    let price = menuItem.getElementsByClassName("price")[0].innerHTML;
+    let imageSrc = menuItem.querySelectorAll(".plate img")[0].src;
+    let id = event.path[0].id;
+    addItemToCart(title, price, imageSrc, id);
+
+    let inCartButtonContents = `
+             <button id=${id} class="in-cart">
+                    <img src="images/check.svg" alt="Check" />
+                    In Cart
+            </button>`;
+    let inCartButton = document.createElement("in-cart");
+    inCartButton.innerHTML = inCartButtonContents;
+    button.parentNode.replaceChild(inCartButton, button);
+
+    updateCartTotal()
+}
+
+function addItemToCart(title, price, imageSrc, id) {
+    var cartRow = document.createElement('li')
```

Comment:
do not use var, onlyuse let or const and a single file should not have more lines of code. move the functions to an other file and call it from there